File: RRComSSys/Projects/CVM/RRComSSys.CVM/src/edu/fiu/cis/rrcomssys/cvm/mte/xcmlmodel/UserDefinition.java
package edu.fiu.cis.rrcomssys.cvm.mte.xcmlmodel;

public class UserDefinition
{
	/* Member Variables */
	private Person _person;
	private IsAttached _isAttached;
	private Device _device;
	
	/* Constructors & Initializers */
	public UserDefinition(Person person, IsAttached isAttached, Device device)
	{
		_person = person;
		_isAttached = isAttached;
		_device = device;
	}
	
	/* Properties */
	public Person getPerson()
	{
		return _person;
	}

	public Device getDevice()
	{
		return _device;
	}

	public IsAttached getIsAttached()
	{
		return _isAttached;
	}

	public boolean isLocal()
	{
		return _device.isLocal();
	}
}

File: RRComSSys/Projects/CVM/RRComSSys.CVM.ObjectModel/XCMLModel/UserDefinition.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Xml.Serialization;

namespace RRComSSys.CVM.ObjectModel.XCMLModel
{
	public class UserDefinition : IXCMLItem
	{
		#region Member Variables
		public Device _device;
		#endregion

		#region Constructors & Initializers
		public UserDefinition() { }
		public UserDefinition(Person person, IsAttached isAttached, Device device)
		{
			this.Person = person;
			this.IsAttached = isAttached;
			this.Device = device;
		}
		#endregion

		#region Properties
		public Person Person { get; set; }
		public IsAttached IsAttached { get; set; }
		public Device Device
		{
			get { return _device; }
			set
			{
				_device = value;
				_device.UserDefinition = this;
			}
		}
		public bool IsLocal
		{
			get
			{
				if (this.Device == null) return false;
				return this.Device.IsLocal;
			}
			set
			{
				this.Device.IsLocal = value;
			}
		}

		public XCMLDocument Document { get; set; }
		#endregion
	}
}
